client/controllers/events/eventFiltration.coffee

```diff
@@ -23,6 +43,12 @@ Template.eventFiltration.onCreated ->
       filters.status =
         $in: status
 
+    selectedLocations = @selectedLocations.find().map (location) ->
+      location.name
+    if selectedLocations.length
+      locationProjection = "locations.#{@countryLevel.get()}"
+      filters[locationProjection] = $in: selectedLocations
```

Comment:
It's an edge case, but It's possible for two separate administrative divisions to share an admin1name or admin2name. The selected location's parent admin name and country should be included in the filter to distinguish between locations that share the same name.